entrypoint/entrypoint/pkg_vuln.py

```diff
@@ -320,3 +311,127 @@ def combine_str_list_into_one_str(str_list: list[str]) -> str:
     if str_element == "":
         str_element = NULL_STR
     return str_element
+
+
+def get_highest_severity_rating(ratings) -> CvssRating:
+    method = get_preferred_vuln_rating_method(ratings)
+    most_severe_rating = get_highest_rating_by_method(method, ratings)
+    cvss = CvssRating()
+    cvss.provider = most_severe_rating["source"]["name"]
+    cvss.severity = most_severe_rating["severity"]
+    if "unknown" in cvss.severity:
+        cvss.severity = "other"
+    cvss.cvss_score = str(most_severe_rating.get("score", 0.0))
+    return cvss
+
+
+class VulnRatingMethod(enum.Enum):
+    CVSSv2 = "CVSSv2"
+    CVSSv3 = "CVSSv3"
+    CVSSv31 = "CVSSv31"
+    CVSSv4 = "CVSSv4"
+    OWASP = "OWASP"
+    SSVC = "SSVC"
+    OTHER = "other"
+
+
+def get_preferred_vuln_rating_method(ratings):
+    if ratings is None:
+        return VulnRatingMethod.OTHER
+
+    found_methods = []
+
+    for rating in ratings:
+        if not rating:
+            continue
+
+        method = rating.get("method", "")
+        if not method:
+            continue
+
+        if is_epss(method, rating):
+            continue
+
+        # we keep a list of each rating method in the
+        # vulnerability so we can present the highest rating
+        # to the end user
+        if method == VulnRatingMethod.CVSSv4.value:
+            found_methods.append(VulnRatingMethod.CVSSv4)
+        elif method == VulnRatingMethod.CVSSv31.value:
+            found_methods.append(VulnRatingMethod.CVSSv31)
+        elif method == VulnRatingMethod.CVSSv3.value:
+            found_methods.append(VulnRatingMethod.CVSSv3)
+        elif method == VulnRatingMethod.CVSSv2.value:
+            found_methods.append(VulnRatingMethod.CVSSv2)
+        elif method == VulnRatingMethod.OWASP.value:
+            found_methods.append(VulnRatingMethod.OWASP)
+        elif method == VulnRatingMethod.SSVC.value:
+            found_methods.append(VulnRatingMethod.SSVC)
+        elif method == VulnRatingMethod.OTHER.value:
+            found_methods.append(VulnRatingMethod.OTHER)
+        else:
+            logging.error(f"Expected a spec-conforming CycloneDX vulnerability rating method, but received '{method}'")
+            continue
+
+    # select method to display to user in priority order
+    if VulnRatingMethod.CVSSv4 in found_methods:
+        return VulnRatingMethod.CVSSv4
+    elif VulnRatingMethod.CVSSv31 in found_methods:
+        return VulnRatingMethod.CVSSv31
+    elif VulnRatingMethod.CVSSv3 in found_methods:
+        return VulnRatingMethod.CVSSv3
+    elif VulnRatingMethod.CVSSv2 in found_methods:
+        return VulnRatingMethod.CVSSv2
+    elif VulnRatingMethod.OWASP in found_methods:
+        return VulnRatingMethod.OWASP
+    elif VulnRatingMethod.SSVC in found_methods:
+        return VulnRatingMethod.SSVC
```

Comment:
How about this? This might be a little simpler.
```suggestion
    rating_method_priority = [VulnRatingMethod.CVSSv4, VulnRatingMethod.CVSSv31, ..., VulnRatingMethod.OTHER]
    for method in method_priority:
        if method in found_methods:
            return method
   return VulnRatingMethod.OTHER
```

Reply:
Good call out, I added this.